### ISSUE TITLE
useless postfix expression isn't useless

### ISSUE THREAD

**Issue by jbaron** [detekt/detekt#7037]
## Expected Behavior
Detekt should not complain about postfix expression being useless if it increases a value that lives outside the scope of the method it is used.

## Observed Behavior
Detekt complaints that the postfix example below is useless, while in fact it does exactly the job it is suppose to do.

## Steps to Reproduce
```kotlin
   private var ID = 0

  /**
   * Generate the next order id
   */
  @Synchronized
  private fun nextId(): String {
      return ID++.toString() // this is the line !!!!!
  }
```

## Context
Not a big issue, just a false negative.

## Your Environment
Maven build
detekt plugin in IntelliJ 2.3.0
Kotlin 1.9.22

**Comment by cortinico** [comment]
This is a valid bug. The problem is here:
https://github.com/detekt/detekt/blob/9cfe9f29276899b435c9eeb42f7631e46120d618/detekt-rules-errorprone/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/UselessPostfixExpression.kt#L69-L79

Basically for return instructions we don't account for properties in the class, which we should.

This is up for grab if someone is looking for an easy contribution
